### PR TITLE
Add flan to the FlexNVMe transport layer

### PR DESCRIPTION
**Pull request by avborup**
The basic building block upon which we can expand our code that uses flan. Currently it only opens and closes a flan pool on a given device. `static` is used to share the same pool across all transport instances, but we should change this down the line.. maybe?

My next step is to write actual objects with flan in the `Write` method.

### CODE REVIEW

File: source/adios2/toolkit/format/bp/BPBase.cpp [hunk 1]
```diff
@@ -368,6 +367,10 @@ BPBase::GetTransportIDs(const std::vector<std::string> &transportsTypes) const
         {
             id = METHOD_POSIX;
         }
+        else if (method == "File_FlexNVMe")
+        {
+            id = METHOD_FLEXNVME;
+        }
```
Comment (albe2669) [inline source/adios2/toolkit/format/bp/BPBase.cpp]: Let's revert all of this file and bpbase.h. We haven't tested it so i don't think it makes any difference.

File: source/adios2/toolkit/format/bp/BPBase.h [hunk 2]
```diff
@@ -376,7 +376,8 @@ class BPBase
         METHOD_FSTREAM = 26,
         METHOD_FILE = 27,
         METHOD_ZMQ = 28,
-        METHOD_MDTM = 29
+        METHOD_MDTM = 29,
+        METHOD_FLEXNVME = 30
```
Comment (albe2669) [inline source/adios2/toolkit/format/bp/BPBase.h]: Revert this file too

File: source/adios2/toolkit/transport/file/FileFlexNVMe.cpp [hunk 3]
```diff
@@ -18,10 +19,11 @@
...
-#include <cstdio>      // remove
-#include <cstring>     // strerror
-#include <errno.h>     // errno
-#include <fcntl.h>     // open
+#include <cstdio>  // remove
+#include <cstring> // strerror
+#include <errno.h> // errno
+#include <fcntl.h> // open
+#include <flan.h>
 #include <sys/stat.h>  // open, fstat
```
Comment (albe2669) [inline source/adios2/toolkit/transport/file/FileFlexNVMe.cpp]: Are we sure we need all of these imports?

File: source/adios2/toolkit/transport/file/FileFlexNVMe.h [hunk 4]
```diff
@@ -30,9 +32,9 @@ class FileFlexNVMe : public Transport
 {
 
 public:
-    FileFlexNVMe(helper::Comm const &comm);
+    explicit FileFlexNVMe(helper::Comm const &comm);
 
-    ~FileFlexNVMe();
+    ~FileFlexNVMe() noexcept;
```
Comment (albe2669) [inline source/adios2/toolkit/transport/file/FileFlexNVMe.h:37]: What does explict and noexept do in this context?

**Review by albe2669 — COMMENTED** [review]
Looks great :1st_place_medal: , there are some comments though.

File: source/adios2/toolkit/transport/file/FileFlexNVMe.h (hunk 4 above)
Comment (avborup) [reply source/adios2/toolkit/transport/file/FileFlexNVMe.h:37]: explicit makes sure that the constructor isnt some type conversion. The linter mostly did this.

noexcept does a compile time check to make sure no exceptions are thrown in the destructor.

File: source/adios2/toolkit/format/bp/BPBase.cpp (hunk 1 above)
Comment (avborup) [reply source/adios2/toolkit/format/bp/BPBase.cpp]: 👍

File: source/adios2/toolkit/transport/file/FileFlexNVMe.cpp (hunk 3 above)
Comment (avborup) [reply source/adios2/toolkit/transport/file/FileFlexNVMe.cpp]: We didn't need them, nice catch. Also removed more unused things 😎

**Review by albe2669 — APPROVED** [review]
Lovely

**Pull request merged**